doc/api/url.md

```diff
@@ -250,7 +250,308 @@ properties of URL objects:
+console.log(myURL.protocol); // "https:"
+console.log(myURL.hostname); // "/example.org"
+console.log(myURL.pathname); // "/foo"
+```
+
+### Class: URL
+#### Constructor: new URL(input[, base])
+
+* `input` {String} The input URL to parse
+* `base` {String | URL} The base URL to resolve against if the `input` is not
+  absolute.
+
+Creates a new `URL` object by parsing the `input` relative to the `base`. If
+`base` is passed as a string, it will be parsed equivalent to `new URL(base)`.
+
+```js
+const myURL = new URL('/foo', 'https://example.org/');
+  // https://example.org/foo
+```
+
+A `TypeError` will be thrown if the `input` or `base` are not valid URLs.
+
+#### url.hash
+
+Gets and sets the fragment portion of the URL.
+
+```js
+const myURL = new URL('http://example.org/foo#bar');
+console.log(myURL.hash);
+  // Prints #bar
+
+myURL.hash = 'baz';
+console.log(myURL.href);
+  // Prints http://example.org/foo#baz
+```
+
+#### url.host
+
+Gets and sets the host portion of the URL.
+
+```js
+const myURL = new URL('http://example.org:81/foo');
+console.log(myURL.host);
+  // Prints example.org:81
+
+myURL.host = 'example.com:82';
+console.log(myURL.href);
+  // Prints http://example.com:82/foo
+```
+
+#### url.hostname
+
+Gets and sets the hostname portion of the URL. The key difference between
+`url.host` and `url.hostname` is that `url.hostname` does *not* include the
+port.
+
+```js
+const myURL = new URL('http://example.org:81/foo');
+console.log(myURL.hostname);
+  // Prints example.org
+
+myURL.hostname = 'example.com:82';
+console.log(myURL.href);
+  // Prints http://example.com:81/foo
+```
+
+#### url.href
+
+Gets and sets the serialized URL.
+
+```js
+const myURL = new URL('http://example.org/foo');
+console.log(myURL.href);
+  // Prints http://example.org/foo
+
+myURL.href = 'https://example.com/bar'
+  // Prints https://example.com/bar
+```
+
+#### url.origin
+
+Gets the read-only Unicode serialization of the URL's origin.
+
+```js
+const myURL = new URL('http://example.org/foo/bar?baz');
+console.log(myURL.origin);
+  // Prints http://exmaple.org
+```
+
+#### url.password
+
+Gets and sets the password portion of the URL.
+
+```js
+const myURL = new URL('http://abc:xyz@example.com');
+console.log(myURL.password);
+  // Prints xyz
+
+myURL.password = '123';
+console.log(myURL.href);
+  // Prints http://abc:123@example.com
+```
+
+#### url.pathname
+
+Gets and sets the path portion of the URL.
+
+```js
+const myURL = new URL('http://example.org/abc/xyz?123');
+console.log(myURL.pathname);
+  // Prints /abc/xyz
+
+myURL.pathname = '/abcdef';
+console.log(myURL.href);
+  // Prints http://example.org/abcdef?123
+```
+
+#### url.port
+
+Gets and sets the port portion of the URL.
+
+```js
+const myURL = new URL('http://example.org:8888');
+console.log(myURL.port);
+  // Prints 8888
+
+myURL.port = 1234;
+console.log(myURL.href);
+  // Prints http://example.org:1234
+```
+
+#### url.protocol
+
+Gets and sets the protocol portion of the URL.
```

Comment:
what if you delete it? actually, for all the properties, what happens if they are deleted?

also, what happens if you assign an invalid value? does it throw, like it does in construction?